### PR TITLE
Assault config

### PR DESCRIPTION
**Pull request by TylerS1066**
Added more options for configuring assaults.

### CODE REVIEW

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/Movecraft.java [hunk 1]
```diff
@@ -190,7 +190,11 @@ public void onEnable() {
         Settings.AssaultDuration = getConfig().getInt("AssaultDuration", 1800);
         Settings.AssaultCostPercent = getConfig().getDouble("AssaultCostPercent", 0.25);
         Settings.AssaultDamagesPerBlock = getConfig().getInt("AssaultDamagesPerBlock", 15);
-        Settings.AssaultRequiredDefendersOnline = getConfig().getInt("AssaultRequiredDefendersOnline", 3);
+        Settings.AssaultRequiredDefendersOnline = getConfig().getInt("AssaultRequiredDefendersOnline", 2);
+        Settings.AssaultRequiredOwnersOnline = getConfig().getInt("AssaultRequiredOwnersOnline", 1);
+        Settings.AssaultMaxBalance = getConfig().getDouble("AssaultMaxBalance", 5000000);
+        Settings.AssaultOwnerWeightPercent = getConfig().getInt("AssaultOwnerWeightPercent", 100);
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/Movecraft.java]: Use decimals for percents

Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/Movecraft.java]: Use decimals for percents

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java [hunk 2]
```diff
@@ -13,46 +13,72 @@
 
     public static boolean areDefendersOnline(ProtectedRegion tRegion) {
         HashSet<UUID> players = new HashSet<>();
-        players.addAll(tRegion.getMembers().getUniqueIds());
         players.addAll(tRegion.getOwners().getUniqueIds());
         int numOnline = 0;
+
+        if(Settings.AssaultRequiredOwnersOnline > 0) {
+            for (UUID playerName : players) {
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java]: change `playerName` to `playerID`

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java [hunk 3]
```diff
@@ -13,46 +13,72 @@
 
     public static boolean areDefendersOnline(ProtectedRegion tRegion) {
         HashSet<UUID> players = new HashSet<>();
-        players.addAll(tRegion.getMembers().getUniqueIds());
         players.addAll(tRegion.getOwners().getUniqueIds());
         int numOnline = 0;
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java]: Use a different variable for owners and members

**Comment by TylerS1066** [comment]
Requested changes done, and sped up areDefendersOnline even more (it now stops once it passes the required number of defenders, and skips each section if it is set to 0).

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java [hunk 4]
```diff
@@ -12,40 +12,54 @@
...
     public static boolean areDefendersOnline(ProtectedRegion tRegion) {
-        HashSet<UUID> players = new HashSet<>();
-        players.addAll(tRegion.getOwners().getUniqueIds());
         int numOnline = 0;
 
         if(Settings.AssaultRequiredOwnersOnline > 0) {
-            for (UUID playerName : players) {
-                if (Bukkit.getPlayer(playerName) != null) {
+            HashSet<UUID> owners = new HashSet<>();
+            owners.addAll(tRegion.getOwners().getUniqueIds());
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java]: This is unnecessary, just use `tRegion.getOwners().getUniqueIds()` for iteration

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java [hunk 5]
```diff
@@ -12,40 +12,54 @@
...
         }
 
-        players.addAll(tRegion.getMembers().getUniqueIds());
-        for (UUID playerName : players) {
-            if (Bukkit.getPlayer(playerName) != null) {
-                numOnline++;
+        numOnline = 0;
+        if(Settings.AssaultRequiredOwnersOnline > 0) {
+            HashSet<UUID> members = new HashSet<>();
+            members.addAll(tRegion.getMembers().getUniqueIds());
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/warfare/assault/AssaultUtils.java]: This is again  unnecessary, just use `tRegion.getMembers().getUniqueIds()` for iteration

**Comment by TylerS1066** [comment]
Ah I see, changes made

**Pull request merged**